js/chat.js

```diff
@@ -0,0 +1,158 @@
+var request;
+
+if(typeof window!="undefined") {
+	// browser
+	request=function(ops,cb) {
+		var x=new XMLHttpRequest();
+		x.onreadystatechange=function() {
+			if(x.readyState !== XMLHttpRequest.DONE)return;
+			cb(null,{statusCode:x.status},JSON.parse(x.responseText));
+		}
+
+		x.open(ops.method,ops.uri);
+		x.setRequestHeader('Content-Type','application/json');
+		x.setRequestHeader('Accept','application/json');
+		x.send(JSON.stringify(ops.json))
+	}
+}
+else {
+	// node
+	request=require('request');
+}
+
+
+
+
+
+
+var API= {
+	__promise_wrap:(endpoint,dat) => {
+		return new Promise( (resolve,reject) => {
+			request({ method: 'POST', uri: 'https://hackmud.com/mobile/'+endpoint+'.json', json:dat},
```

Comment:
change base url to 
https://www.hackmud.com/mobile/
cert is signed on www. so the servers will redirect.